home/.chezmoiscripts/universal/run_onchange_before_11-install-wazuh-agent.sh.tmpl

```diff
@@ -0,0 +1,141 @@
+{{- if ne .host.distro.family "windows" -}}
+#!/usr/bin/env bash
+# @file Wazuh agent installation and registration
+# @brief Ensures Wazuh Agent is installed and the agent joined to the given Wazuh Manager
+# @description
+#     This script ensures Wazuh Agent is installed and joined to Wazuh Manager under the `default` group. If Wazuh Manager IP is not provided
+#     in the input, agent installation is skipped. On MacOS and Archlinux, the script builds the agent and then installs.
+#
+#     No configuraiton other than the Manager IP is added to the config file by this script. Please refer the
+#     [Deployment variables document](https://documentation.wazuh.com/current/user-manual/deployment-variables/deployment-variables.html),
+#     relevant to the OS you are using for more information.
+#
+#     ## Configuration Variables
+#
+#     The following chart details the input variable(s) that are used to determine the configuration of the Wazuh Agent:
+#
+#     | Variable        | Description                                                |
+#     |-----------------|------------------------------------------------------------|
+#     | `wazuh.manager` | The IP address where Wazuh Manager is available            |
+
+{{ includeTemplate "universal/profile-before" }}
+{{ includeTemplate "universal/logg-before" }}
+
+### Install Wazuh Agent
+if [ -n "{{ .host.wazuh.manager }}" ]; then
+    if [ -d /Applications ] && [ -d /System ]; then
+        # macOS
+        logg info 'Installing build dependencies'
+        brew install automake autoconf libtool cmake jq
+        logg info 'Finding the latest version of Wazuh Agent'
+        WAZUH_AGENT_VERSION=$(curl -s https://api.github.com/repos/wazuh/wazuh/releases/latest | jq -r '.tag_name')
+        logg info 'Downloading the latest version of Wazuh Agent'
+        cd /tmp
+        curl -Ls "https://github.com/wazuh/wazuh/archive/$WAZUH_AGENT_VERSION.tar.gz" | tar zx
+        logg info 'Installing the latest version of Wazuh Agent'
+        cd wazuh*
+        tee -a etc/preloaded-vars.conf << EOF
+USER_LANGUAGE="en"
+USER_NO_STOP="y"
+USER_INSTALL_TYPE="agent"
+USER_DIR="/Library/Ossec"
+USER_ENABLE_SYSCHECK="y"
+USER_ENABLE_ROOTCHECK="y"
+USER_ENABLE_OPENSCAP="y"
+USER_ENABLE_ACTIVE_RESPONSE="y"
+USER_AGENT_SERVER_IP="{{ .host.wazuh.manager }}"
+USER_CA_STORE="n"
+EOF
+        sudo ./install.sh
+        sudo /Library/Ossec/bin/wazuh-control start
```

Comment:
Will this run it on reboot?

Reply:
Yes, the script install a Daemon and loads it.